homeassistant/components/history_stats/sensor.py

```diff
@@ -322,13 +379,6 @@ def update_period(self):
         if end is None:
             end = start + self._duration
 
-        if start > dt_util.now():
```

Comment:
We still need to update the period or have the problem in https://github.com/home-assistant/core/issues/26857